### ISSUE TITLE
Move breath system into MTG

### ISSUE THREAD

**Issue by Zughy** [minetest/minetest#14125]
### Problem

Breath is a legacy feature we've been carrying due to MT wanting to be like Minecraft in the early years of Minetest.  We don't have hunger, why should we have breath? Two major problems:
1. a lot of games don't need breath, and the ones that do they can simply implement it [on their own](https://github.com/minetest/minetest/issues/10098#issuecomment-648781440)
2. the current system is pretty hardcoded. Quoting @hecktest: "What business does the engine have deciding where my character's head actually is? Depending on the animation, the head's position might be off by as much as 2 nodes" (from https://github.com/minetest/minetest/issues/10098#issuecomment-648964131). Or, another example, I can't choose where to put the breath bar, getting in the way with my HUDs

![image](https://github.com/minetest/minetest/assets/63455151/f939f3e8-3de8-4604-a0ec-2cf74d5c05f8)

### Solutions

Get rid of it from the engine, featuring it as a mod of MTG

### Alternatives

Add more features, complicating your life

**Comment by rubenwardy** [comment]
To work around the breaking changes, we might wish to move to builtin Lua first and then move to MTG in 6.0

**Comment by Zughy** [comment]
@rubenwardy is it that breaking of a change if people just need to update MTG?

**Comment by rubenwardy** [comment]
Yes, because you're removing an API feature from the engine. This will break games and mods that rely on it

**Comment by rubenwardy** [comment]
It can be a mod in MTG that's also available externally, like sfinv
